source/common/http/http1/codec_impl.cc

```diff
@@ -379,12 +383,14 @@ void RequestEncoderImpl::encodeHeaders(const RequestHeaderMap& headers, bool end
   const HeaderEntry* host = headers.Host();
   bool is_connect = HeaderUtility::isConnect(headers);
 
-  if (!method || (!path && !is_connect)) {
-    // TODO(#10878): This exception does not occur during dispatch and would not be triggered under
-    // normal circumstances since inputs would fail parsing at ingress. Replace with proper error
-    // handling when exceptions are removed. Include missing host header for CONNECT.
-    throw CodecClientException(":method and :path must be specified");
-  }
+  // TODO(#10878): Include missing host header for CONNECT.
+  // The RELEASE_ASSERT below does not change the existing behavior of `encodeHeaders`.
+  // The `encodeHeaders` used to throw on errors. Callers of `encodeHeaders()` do not catch
+  // exceptions and this would cause abnormal process termination in error cases. This change
+  // replaces abnormal process termination from unhandled exception with the RELEASE_ASSERT. Further
+  // work will replace this RELEASE_ASSERT with proper error handling.
+  RELEASE_ASSERT(method && (path || is_connect), ":method and :path must be specified");
```

Comment:
Was it possible for this call to encodeHeaders to ever happen under the dispatch call stack?

It may be worth an ASSERT(!dispatching_) to make that explicit.

Reply:
Done. Shouldn't be. The response encoder encodeHeaders will be though.